internal/cmd/commands/authenticate/password.go

```diff
@@ -74,6 +74,13 @@ func (c *PasswordCommand) Flags() *base.FlagSets {
 		Usage:  "The auth-method resource to use for the operation.",
 	})
 
+	f.StringVar(&base.StringVar{
+		Name:   "scope-id",
+		EnvVar: "BOUNDARY_SCOPE_ID",
+		Target: &c.FlagScopeId,
+		Usage:  "The scope ID to use for the operation.",
+	})
```

Comment:
I would think with the addition of this we should also add an error if the auth-method-id is not in the scope-id specified, since it may confuse users who provide conflicting information. What do you think?

Reply:
That's a great point, given the order of operations:
1. If the scope id and AM id are not present, it defaults to global scope and if there is no primary in global scope, we should have a specific error for that.
2. If the scope id is pass, and no primary is found, an error specific to no primary being found in that scope id should be emitted.

Reply:
What if there is a primary auth method ID but it's not a password auth method 🤔?

Reply:
I'm not sure what that would do, but the PR here wouldn't change the behavior since that could occur today.

Reply:
I.e., you could do a `boundary authenticate password` and pass a OIDC AM ID to the -auth-method-id flag.

Reply:
True, I guess I only think we need a new error for when the scope-id and auth-method-id conflict. I'm not sure if you agreed in your first reply, but I think it's important to avoid ambiguity for the user.